cardano-api-gen/src/Gen/Cardano/Api/Typed.hs

```diff
@@ -577,6 +578,18 @@ genTxBodyContent era = do
     , txMintValue = mintValue
     }
 
+genTxInsCollateral :: CardanoEra era -> Gen (TxInsCollateral era)
+genTxInsCollateral era =
+  case era of
+    ByronEra -> pure TxInsCollateralNone
+    ShelleyEra -> pure TxInsCollateralNone
+    AllegraEra -> pure TxInsCollateralNone
+    MaryEra -> pure TxInsCollateralNone
+    AlonzoEra -> Gen.choice
+      [ pure TxInsCollateralNone
+      , TxInsCollateral CollateralInAlonzoEra <$> Gen.list (Range.linear 0 10) genTxIn
+      ]
```

Comment:
We should use the "feature supported in era" pattern for these things, rather than enumerating all the eras.
```suggestion
    case collateralSupportedInEra era of
      Nothing        -> pure TxInsCollateralNone
      Just supported -> Gen.choice
                          [ pure TxInsCollateralNone
                          , TxInsCollateral supported <$> Gen.list (Range.linear 0 10) genTxIn
                          ]
```